### ISSUE TITLE
Cloud Image Name erroneously not accepted (blocked by regex check)

### ISSUE THREAD

**Issue by amolari** [f5devcentral/f5-bigip-image-generator#32]
**Describe the bug**
When configuring the following
`CLOUD_IMAGE_NAME: "BIGIP-14.1.4.4.0.27.4-CB01-BYOL-All-Modules-2Boot-Loc"`
I get the error:
```
Successfully extracted '/root/f5-bigip-image-generator-1.15/src/lib/bash/../../../artifacts/BIGIP-14.1.4.4-0.0.4/alibaba/all_2slot/ve.info.json' from '/root/source/BIGIP-14.1.4.4-0.0.4.iso'.
Traceback (most recent call last):
  File "/root/f5-bigip-image-generator-1.15/src/bin/prepare_image.py", line 78, in <module>
    main()
  File "/root/f5-bigip-image-generator-1.15/src/bin/prepare_image.py", line 55, in main
    ImageController.check_valid_name(args.platform, args.user_image_name)
  File "/root/f5-bigip-image-generator-1.15/src/lib/python/image/image_controller.py", line 120, in check_valid_name
    cloud_image_name.check_valid_name(user_image_name)
  File "/root/f5-bigip-image-generator-1.15/src/lib/python/image/image_name.py", line 138, in check_valid_name
    self.rules.match_regex, name))
ValueError: Found unexpected characters not matching ^(?!(https://|http://))[a-zA-Z][a-zA-Z0-9\-_]+$ in BIGIP-14.1.4.4.0.27.4-CB01-BYOL-All-Modules-2Boot-Loc
```

If i comment that setting, I get some local image named `Hotfix-BIGIP-14.1.4.4-0.27.4.ALL.alibaba.tar.gz`

**To Reproduce**
v1.15
In my case, Platform=Alibaba

**Expected behavior**
Setting a cloud image name to "BIGIP-14.1.4.4.0.27.4-CB01-BYOL-All-Modules-2Boot-Loc" should be possible, as it's inline with
cloud provider name restriction.

**Additional context**

**Comment by kmaloney** [comment]
I can repro this.  The regex doesn't include the period character.  I also think that the https/http is some artifact of an earlier implementation that should be removed.  This issue is being tracked internally with VECLOUD-1272.

Proposed change if you want to try the fix:

```
$ git diff
diff --git a/src/lib/python/image/alibaba_image_name.py b/src/lib/python/image/alibaba_image_name.py
index e0003144..cf3b4910 100644
--- a/src/lib/python/image/alibaba_image_name.py
+++ b/src/lib/python/image/alibaba_image_name.py
@@ -1,5 +1,5 @@
 """alibaba_image_name module"""
-# Copyright (C) 2019 F5 Networks, Inc
+# Copyright (C) 2019-2021 F5 Networks, Inc
 #
 # Licensed under the Apache License, Version 2.0 (the "License"); you may not
 # use this file except in compliance with the License. You may obtain a copy of
@@ -31,8 +31,7 @@ class AlibabaImageName(ImageName):

         min_chars = int(get_config_value('ALIBABA_IMAGE_NAME_LENGTH_MIN'))
         max_chars = int(get_config_value('ALIBABA_IMAGE_NAME_LENGTH_MAX'))
-        rules = ImageNameRules(min_chars, max_chars,
-                               match_regex=r'^(?!(https://|http://))[a-zA-Z][a-zA-Z0-9\-_]+$')
+        rules = ImageNameRules(min_chars, max_chars, match_regex=r'^[a-zA-Z][a-zA-Z0-9\.\-_]+$')

         transform = ImageNameTransform(disallowed_regex=r'[^a-zA-Z0-9\-_]', replacement_char='-',
                                        to_lower=False)
```

**Comment by amolari** [comment]
@kmaloney I've tested your proposed change and it works.

**Comment by kmaloney** [comment]
Great.  Thanks for confirming this.

**Comment by kmaloney** [comment]
Resolved in v1.16.
